src/main/java/ui/components/FilterTextField.java

```diff
@@ -89,14 +99,15 @@ private void setup(Predicate<String> validation) {
         });
         setOnKeyReleased(e -> {
             e.consume();
-            if (e.getCode() == KeyCode.ENTER) {
-                confirmEdit();
-            } else if (e.getCode() == KeyCode.ESCAPE) {
-                if (getText().equals(previousText)) {
-                    onCancel.run();
-                } else {
-                    revertEdit();
-                }
+
+            if (e.getCode() == KeyCode.ENTER) handleOnEnter();
+
+            if (e.getCode() == KeyCode.ESCAPE) handleOnEscape();
```

Comment:
We could use `else if` here, to make it clear that the effects of the key handlers are mutually exclusive.